parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -570,22 +571,30 @@ def add_to_new_modules(new_module: dict):
                 module_temp['compilation'] = new_module.get('compilation-status', 'PENDING')
                 module_temp['date'] = date
                 module_temp['schema'] = new_module['schema']
-                mod_details = [module_temp]
+                mod_details: t.List[t.Dict[str, str]] = [module_temp]
 
                 # Loop through all other available revisions of the module
-                for mod in [revision for name in data.values() for revision in name.values()]:
-                    module_temp = {}
-                    revision = mod['revision']
-                    if revision == new_module['revision']:
+                revision = ''
+                try:
+                    for mod in [revision for name in data.values() for revision in name.values()]:
+                        module_temp = {}
+                        revision = mod['revision']
+                        if revision == new_module['revision']:
+                            continue
+                        module_temp['revision'] = revision
+                        module_temp['date'] = get_revision_datetime(mod)
+                        module_temp['name'] = name
+                        module_temp['organization'] = mod['organization']
+                        module_temp['schema'] = mod['schema']
+                        module_temp['compilation'] = mod.get('compilation-status', 'PENDING')
+                        module_temp['semver'] = mod['derived-semantic-version']
+                        mod_details.append(module_temp)
+                except KeyError as e:
+                    LOGGER.error('Existing module {}@{} is missing the {} field'.format(name, revision, e))
+                    if str(e) == 'derived-semantic-version':
+                        LOGGER.error('Cannot resolve semver for {}@{}'.format(name, new_revision))
                         continue
-                    module_temp['revision'] = revision
-                    module_temp['date'] = get_revision_datetime(mod)
-                    module_temp['name'] = name
-                    module_temp['organization'] = mod.get('organization')
-                    module_temp['schema'] = mod.get('schema')
-                    module_temp['compilation'] = mod.get('compilation-status', 'PENDING')
-                    module_temp['semver'] = mod.get('derived-semantic-version')
-                    mod_details.append(module_temp)
+
```

Comment:
It is intentionally inside the if, we can find the semver for the new module even if we don't have the organization or schema for the older version of the module. What I forgot to do, is change the order so that the semver is assigned first.